File: problems/459-Repeated-Substring-Pattern/solution.go
func check(s string, t string) bool {
	if len(s) == len(t) {
		for i := 0; i < len(s); i++ {
			if s[i] != t[i] {
				return false
			}
		}
		return true
	}
	for i := 0; i < len(t)/len(s); i++ {
		if !check(s, string(t[i*len(s):(i+1)*len(s)])) {
			return false
		}
	}
	return true
}

func repeatedSubstringPattern(s string) bool {
	for i := 1; i <= len(s)/2; i++ {
		if len(s)%i == 0 {
			if check(string(s[:i]), s) {
				return true
			}
		}
	}

	return false
}

File: problems/459-Repeated-Substring-Pattern/solution.java
class Solution {
    public boolean repeatedSubstringPattern(String s) {
        for (int i = 1; i < s.length(); i++) {
            if (s.length() % i != 0) {
                continue;
            }

            boolean flag = true;
            for (int j = 1; j * i < s.length(); j++) {
                if (!s.substring(0, i).equals(s.substring(j * i, (j + 1) * i))) {
                    flag = false;
                    break;
                }
            }

            if (flag) {
                return true;
            }
        }

        return false;
    }
}

